doc/cookbook/source/examples/classifier/hierarchical.rst

```diff
@@ -0,0 +1,36 @@
+=======================
+Hierarchical Clustering
+=======================
+
+Hierarchical clustering is a method of cluster analysis which seeks to build a hierarchy of clusters.
+In Shogun, we applies "bottom up" approach for the hierarchical clustering: each observation starts in its own cluster, and pairs of clusters are merged as one moves up the hierarchy.
+
+The merges are determined in a greedy manner. 
+To start with, we construct a distance matrix, where it saves the distance between the i-th and j-th elements.
+For each step, the element pair that falls into different clusters with the shortest distance is merged, by assigning the pair to the same cluster. 
+
+-------
+Example
+-------
+
+Imagine we have files with the training data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) as:
+
+.. sgexample:: hierarchical.sg:create_features
+
+In order to run :sgclass:`CHierarchical`, we need to choose a distance, for example :sgclass:`CEuclideanDistance`, or other sub-classes of :sgclass:`CDistance`. The distance is initialized with the data we want to classify.
+
+.. sgexample:: hierarchical.sg:choose_distance
+
+We then create an instance of the :sgclass:`CHierarchical` classifier by passing it `merges` and `CDistance`.
+`Merges` assigns the steps of merging we expect to have in the training. 
+
+.. sgexample:: hierarchical.sg:create_instance
+
+And we can extract the information of the two merged elements, as while as the distance between them in each merging step:
+
+.. sgexample:: hierarchical.sg:extract_results
```

Comment:
if you put this example in "classifier" please also show how to extract predicted labels